ui/views.py

```diff
@@ -103,50 +133,72 @@ class VideoEmbed(TemplateView):
     """Display embedded video"""
     template_name = 'ui/video_embed.html'
 
-    def get_context_data(self, video_key, **kwargs):  # pylint: disable=arguments-differ
+    def get(self, request, *args, **kwargs):
+        self.video = get_object_or_404(Video, key=kwargs['video_key'])
+        return conditional_response(self, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        video = get_object_or_404(Video, key=video_key)
-        if not ui_permissions.has_video_view_permission(video, self.request):
-            raise PermissionDenied
-        context['video'] = video
+        context['video'] = self.video
         context["js_settings_json"] = json.dumps({
             **default_js_settings(self.request),
-            'video': VideoSerializer(video).data,
+            'video': VideoSerializer(self.video).data,
         })
         return context
 
 
-class TechTVDetail(VideoDetail):
+class TechTVDetail(TemplateView):
     """
     Video detail page for a TechTV-based URL
     """
-    def get_context_data(self, video_key, **kwargs):
-        # There might be more than one imported TechTV video with this id
-        ttv_videos = get_list_or_404(TechTVVideo.objects.filter(ttv_id=video_key))
-        video = ttv_videos[0].video
-        return super().get_context_data(video.hexkey, **kwargs)
+    template_name = "ui/video_detail.html"
+
+    def get(self, request, *args, **kwargs):
+        ttv_videos = get_list_or_404(TechTVVideo.objects.filter(ttv_id=kwargs['video_key']))
```

Comment:
Is it possible for there to be multiple `TechTVVideo` objects with the same `ttv_id`?

Reply:
yes, if a video is part of 2+ collections there will be a `TechTVVideo` object for each, with the same `ttv_id`